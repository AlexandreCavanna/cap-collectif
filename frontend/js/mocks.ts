/* eslint-env jest */
import type { IntlShape, MessageDescriptor } from 'react-intl'
import 'react-intl'
import type { RelayPaginationProp, RelayRefetchProp } from 'react-relay'

export const $refType: any = null
export const $fragmentRefs: any = null
export const intlMock: IntlShape = {
  locale: 'fr-FR',
  formats: {},
  messages: {},
  now: () => 0,
  formatHTMLMessage: (message: MessageDescriptor) => String(message),
  formatPlural: (message: string) => String(message),
  formatNumber: (message: string) => String(message),
  formatRelative: (message: string) => String(message),
  formatTime: (message: string) => String(message),
  formatDate: (message: string) => String(message),
  formatMessage: (message: MessageDescriptor) => String(message.id),
}
export const formMock: ReduxFormFormProps = {
  anyTouched: false,
  array: {
    insert: jest.fn(),
    move: jest.fn(),
    pop: jest.fn(),
    push: jest.fn(),
    remove: jest.fn(),
    removeAll: jest.fn(),
    shift: jest.fn(),
    splice: jest.fn(),
    swap: jest.fn(),
    unshift: jest.fn(),
  },
  asyncValidate: jest.fn(),
  asyncValidating: false,
  autofill: jest.fn(),
  blur: jest.fn(),
  change: jest.fn(),
  clearAsyncError: jest.fn(),
  clearSubmit: jest.fn(),
  destroy: jest.fn(),
  dirty: false,
  dispatch: jest.fn(),
  error: null,
  form: 'formName',
  handleSubmit: jest.fn(),
  initialize: jest.fn(),
  initialized: true,
  // initialValues: {},
  invalid: false,
  pristine: true,
  reset: jest.fn(),
  submitting: false,
  submitFailed: false,
  submitSucceeded: false,
  touch: jest.fn(),
  untouch: jest.fn(),
  updateSyncErrors: jest.fn(),
  valid: true,
  warning: null,
}
const environment = {
  UNSTABLE_getDefaultRenderPolicy: jest.fn(),
  __log: jest.fn(),
  getOperationTracker: jest.fn(),
  isRequestActive: jest.fn(),
  isServer: jest.fn(),
  options: jest.fn(),
  applyMutation: jest.fn(),
  sendMutation: jest.fn(),
  lookup: jest.fn(),
  sendQuery: jest.fn(),
  subscribe: jest.fn(),
  streamQuery: jest.fn(),
  retain: jest.fn(),
  unstable_internal: {
    areEqualSelectors: jest.fn(),
    createFragmentSpecResolver: jest.fn(),
    createOperationDescriptor: jest.fn(),
    getDataIDsFromFragment: jest.fn(),
    getDataIDsFromObject: jest.fn(),
    getFragment: jest.fn(),
    getPluralSelector: jest.fn(),
    getRequest: jest.fn(),
    getSelector: jest.fn(),
    getSelectorsFromObject: jest.fn(),
    getSingularSelector: jest.fn(),
    getVariablesFromFragment: jest.fn(),
    getVariablesFromObject: jest.fn(),
    getVariablesFromPluralFragment: jest.fn(),
    getVariablesFromSingularFragment: jest.fn(),
    isFragment: jest.fn(),
    isRequest: jest.fn(),
  },
  applyUpdate: jest.fn(),
  check: jest.fn(),
  commitPayload: jest.fn(),
  commitUpdate: jest.fn(),
  execute: jest.fn(),
  executeMutation: jest.fn(),
  executeWithSource: jest.fn(),
  getNetwork: jest.fn(),
  getStore: jest.fn(),
  areEqualSelectors: jest.fn(),
  createFragmentSpecResolver: jest.fn(),
  createOperationDescriptor: jest.fn(),
  getDataIDsFromFragment: jest.fn(),
  getDataIDsFromObject: jest.fn(),
  getFragment: jest.fn(),
  getPluralSelector: jest.fn(),
  getRequest: jest.fn(),
  getSelector: jest.fn(),
  getSelectorsFromObject: jest.fn(),
  getSingularSelector: jest.fn(),
  getVariablesFromFragment: jest.fn(),
  getVariablesFromObject: jest.fn(),
  getVariablesFromPluralFragment: jest.fn(),
  requiredFieldLogger: jest.fn(),
  revertUpdate: jest.fn(),
  replaceUpdate: jest.fn(),
}
export const relayPaginationMock: RelayPaginationProp = {
  environment,
  hasMore: () => false,
  isLoading: () => false,
  loadMore: jest.fn(),
  refetchConnection: jest.fn(),
}
export const relayRefetchMock: RelayRefetchProp = {
  environment,
  refetch: jest.fn(),
}
export const googleAddressMock = {
  lat: 43.6424564,
  lng: -79.3755156,
  json: '[{"formatted_address":"10 Yonge St, Toronto, ON M5E 1R4, Canada","geometry":{"location":{"lat":43.6424564,"lng":-79.3755156},"location_type":"ROOFTOP","viewport":{"south":43.6411074197085,"west":-79.37686458029151,"north":43.6438053802915,"east":-79.37416661970849}},"types":["street_address"],"address_components":[{"long_name":"10","short_name":"10","types":["street_number"]},{"long_name":"Yonge Street","short_name":"Yonge St","types":["route"]},{"long_name":"Old Toronto","short_name":"Old Toronto","types":["political","sublocality","sublocality_level_1"]},{"long_name":"Toronto","short_name":"Toronto","types":["locality","political"]},{"long_name":"Toronto Division","short_name":"Toronto Division","types":["administrative_area_level_2","political"]},{"long_name":"Ontario","short_name":"ON","types":["administrative_area_level_1","political"]},{"long_name":"Canada","short_name":"CA","types":["country","political"]},{"long_name":"M5E 1R4","short_name":"M5E 1R4","types":["postal_code"]}],"place_id":"ChIJrbgtxSvL1IkR2xGe7JZyuw0","plus_code":{"compound_code":"JJRF+XQ Toronto, ON, Canada","global_code":"87M2JJRF+XQ"}}]',
}
