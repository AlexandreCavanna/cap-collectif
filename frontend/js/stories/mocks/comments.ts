import { author } from './users'

export const comment = {
  id: 'comment1',
  user: author,
  trashedStatus: '',
  createdAt: '1 mars 2018',
  publishedAt: '1 mars 2018',
  published: true,
  pinned: true,
  body: 'Lorem ipsum dolor amet direct trade cornhole exercitation hashtag mixtape, master cleanse vexillologist. Irure kickstarter single-origin coffee fashion axe vexillologist viral. Selfies etsy cloud bread heirloom, sunt craft beer fixie palo santo asymmetrical.',
  votes: {
    totalCount: 50,
  },
  viewerHasVote: false,
  contribuable: true,
  reported: false,
}
export const comments = [
  { ...comment, id: 'comment1-2' },
  { ...comment, id: 'comment1-3', highlighted: true },
  { ...comment, id: 'comment1-7', trashedStatus: 'INVISIBLE', body: null },
  { ...comment, id: 'comment1-4', pinned: false },
  { ...comment, id: 'comment1-5', pinned: false, published: false },
  { ...comment, id: 'comment1-6', user: { ...comment.user, isViewer: true } },
]
