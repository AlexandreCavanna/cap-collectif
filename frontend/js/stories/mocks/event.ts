import { author } from './users'

export const event = {
  title:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi',
  url: '#',
  media: {
    url: 'https://picsum.photos/300/400',
  },
  timeRange: {
    startAt: '2030-03-10 00:00:00',
    endAt: '2032-06-15 00:00:00',
  },
  googleMapsAddress: {
    json: '[{"address_components":[{"long_name":"111","short_name":"111","types":["street_number"]},{"long_name":"Avenue Jean Jaurès","short_name":"Avenue Jean Jaurès","types":["route"]},{"long_name":"Lyon","short_name":"Lyon","types":["locality","political"]},{"long_name":"Rhône","short_name":"Rhône","types":["administrative_area_level_2","political"]},{"long_name":"Auvergne-Rhône-Alpes","short_name":"Auvergne-Rhône-Alpes","types":["administrative_area_level_1","political"]},{"long_name":"France","short_name":"FR","types":["country","political"]},{"long_name":"69007","short_name":"69007","types":["postal_code"]}],"formatted_address":"111 Avenue Jean Jaurès, 69007 Lyon, France","geometry":{"location":{"lat":45.742842,"lng":4.84068000000002},"location_type":"ROOFTOP","viewport":{"south":45.7414930197085,"west":4.839331019708538,"north":45.74419098029149,"east":4.842028980291502}},"place_id":"ChIJHyD85zjq9EcR8Yaae-eQdeQ","plus_code":{"compound_code":"PRVR+47 Lyon, France","global_code":"8FQ6PRVR+47"},"types":["street_address"]}]',
  },
  themes: [
    {
      id: '1',
      title: 'Immobilier',
    },
    {
      id: '2',
      title: 'Justice',
    },
  ],
  author,
}
export const events = [
  {
    title:
      'Lorem ipsum dolor sit amet, et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi',
    url: '#',
    media: {
      url: 'https://picsum.photos/300/400',
    },
    timeRange: {
      startAt: '2030-03-10 00:00:00',
      endAt: '2032-06-15 00:00:00',
    },
    googleMapsAddress: {
      json: '[{"address_components":[{"long_name":"111","short_name":"111","types":["street_number"]},{"long_name":"Avenue Jean Jaurès","short_name":"Avenue Jean Jaurès","types":["route"]},{"long_name":"Lyon","short_name":"Lyon","types":["locality","political"]},{"long_name":"Rhône","short_name":"Rhône","types":["administrative_area_level_2","political"]},{"long_name":"Auvergne-Rhône-Alpes","short_name":"Auvergne-Rhône-Alpes","types":["administrative_area_level_1","political"]},{"long_name":"France","short_name":"FR","types":["country","political"]},{"long_name":"69007","short_name":"69007","types":["postal_code"]}],"formatted_address":"111 Avenue Jean Jaurès, 69007 Lyon, France","geometry":{"location":{"lat":45.742842,"lng":4.84068000000002},"location_type":"ROOFTOP","viewport":{"south":45.7414930197085,"west":4.839331019708538,"north":45.74419098029149,"east":4.842028980291502}},"place_id":"ChIJHyD85zjq9EcR8Yaae-eQdeQ","plus_code":{"compound_code":"PRVR+47 Lyon, France","global_code":"8FQ6PRVR+47"},"types":["street_address"]}]',
    },
    themes: [
      {
        id: '1',
        title: 'Immobilier',
      },
      {
        id: '2',
        title: 'Justice',
      },
    ],
    author,
  },
  {
    title:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi',
    url: '#',
    media: {
      url: 'https://picsum.photos/300/400',
    },
    timeRange: {
      startAt: '2030-03-10 00:00:00',
      endAt: '2032-06-15 00:00:00',
    },
    googleMapsAddress: {
      json: '[{"address_components":[{"long_name":"111","short_name":"111","types":["street_number"]},{"long_name":"Avenue Jean Jaurès","short_name":"Avenue Jean Jaurès","types":["route"]},{"long_name":"Lyon","short_name":"Lyon","types":["locality","political"]},{"long_name":"Rhône","short_name":"Rhône","types":["administrative_area_level_2","political"]},{"long_name":"Auvergne-Rhône-Alpes","short_name":"Auvergne-Rhône-Alpes","types":["administrative_area_level_1","political"]},{"long_name":"France","short_name":"FR","types":["country","political"]},{"long_name":"69007","short_name":"69007","types":["postal_code"]}],"formatted_address":"111 Avenue Jean Jaurès, 69007 Lyon, France","geometry":{"location":{"lat":45.742842,"lng":4.84068000000002},"location_type":"ROOFTOP","viewport":{"south":45.7414930197085,"west":4.839331019708538,"north":45.74419098029149,"east":4.842028980291502}},"place_id":"ChIJHyD85zjq9EcR8Yaae-eQdeQ","plus_code":{"compound_code":"PRVR+47 Lyon, France","global_code":"8FQ6PRVR+47"},"types":["street_address"]}]',
    },
    themes: [
      {
        id: '1',
        title: 'Immobilier',
      },
      {
        id: '2',
        title: 'Justice',
      },
    ],
    author,
  },
  {
    title: 'Lorem ipsum dolor sit amet',
    url: '#',
    media: {
      url: 'https://picsum.photos/300/400',
    },
    timeRange: {
      startAt: '2030-03-10 00:00:00',
      endAt: '2032-06-15 00:00:00',
    },
    googleMapsAddress: {
      json: '[{"address_components":[{"long_name":"111","short_name":"111","types":["street_number"]},{"long_name":"Avenue Jean Jaurès","short_name":"Avenue Jean Jaurès","types":["route"]},{"long_name":"Lyon","short_name":"Lyon","types":["locality","political"]},{"long_name":"Rhône","short_name":"Rhône","types":["administrative_area_level_2","political"]},{"long_name":"Auvergne-Rhône-Alpes","short_name":"Auvergne-Rhône-Alpes","types":["administrative_area_level_1","political"]},{"long_name":"France","short_name":"FR","types":["country","political"]},{"long_name":"69007","short_name":"69007","types":["postal_code"]}],"formatted_address":"111 Avenue Jean Jaurès, 69007 Lyon, France","geometry":{"location":{"lat":45.742842,"lng":4.84068000000002},"location_type":"ROOFTOP","viewport":{"south":45.7414930197085,"west":4.839331019708538,"north":45.74419098029149,"east":4.842028980291502}},"place_id":"ChIJHyD85zjq9EcR8Yaae-eQdeQ","plus_code":{"compound_code":"PRVR+47 Lyon, France","global_code":"8FQ6PRVR+47"},"types":["street_address"]}]',
    },
    themes: [
      {
        id: '1',
        title: 'Immobilier',
      },
      {
        id: '2',
        title: 'Justice',
      },
    ],
    author,
  },
  {
    title:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi',
    url: '#',
    media: {
      url: 'https://picsum.photos/300/400',
    },
    timeRange: {
      startAt: '2030-03-10 00:00:00',
      endAt: '2032-06-15 00:00:00',
    },
    googleMapsAddress: {
      json: '[{"address_components":[{"long_name":"111","short_name":"111","types":["street_number"]},{"long_name":"Avenue Jean Jaurès","short_name":"Avenue Jean Jaurès","types":["route"]},{"long_name":"Lyon","short_name":"Lyon","types":["locality","political"]},{"long_name":"Rhône","short_name":"Rhône","types":["administrative_area_level_2","political"]},{"long_name":"Auvergne-Rhône-Alpes","short_name":"Auvergne-Rhône-Alpes","types":["administrative_area_level_1","political"]},{"long_name":"France","short_name":"FR","types":["country","political"]},{"long_name":"69007","short_name":"69007","types":["postal_code"]}],"formatted_address":"111 Avenue Jean Jaurès, 69007 Lyon, France","geometry":{"location":{"lat":45.742842,"lng":4.84068000000002},"location_type":"ROOFTOP","viewport":{"south":45.7414930197085,"west":4.839331019708538,"north":45.74419098029149,"east":4.842028980291502}},"place_id":"ChIJHyD85zjq9EcR8Yaae-eQdeQ","plus_code":{"compound_code":"PRVR+47 Lyon, France","global_code":"8FQ6PRVR+47"},"types":["street_address"]}]',
    },
    themes: [
      {
        id: '1',
        title: 'Immobilier',
      },
      {
        id: '2',
        title: 'Justice',
      },
    ],
    author,
  },
]
