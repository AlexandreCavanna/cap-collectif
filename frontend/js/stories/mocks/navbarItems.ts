export const items = [
  {
    id: 12,
    title: 'Accueil',
    link: '/',
    hasEnabledFeature: true,
    children: [],
    active: true,
  },
  {
    id: 5,
    title: 'Actualités',
    link: '/blog',
    hasEnabledFeature: true,
    children: [],
    active: false,
  },
  {
    id: 11,
    title: 'Thèmes',
    link: '/themes',
    hasEnabledFeature: true,
    children: [],
    active: false,
  },
  {
    id: 4,
    title: 'Évènements',
    link: '/events',
    hasEnabledFeature: true,
    children: [],
    active: false,
  },
  {
    id: 10,
    title: 'Projets participatifs',
    link: '/projects',
    hasEnabledFeature: true,
    children: [],
    active: false,
  },
  {
    id: 23,
    title: 'Custom page',
    link: '/custom-page',
    hasEnabledFeature: true,
    children: [],
    active: false,
  },
  {
    id: 6,
    title: 'Comment ça marche',
    link: '/pages/comment-%C3%A7a-marche',
    hasEnabledFeature: true,
    children: [],
    active: false,
  },
]
export const itemWithChildren = {
  id: 9,
  title: 'À propos',
  link: '',
  hasEnabledFeature: true,
  children: [
    {
      id: 8,
      title: 'Contact',
      link: '/contact',
      hasEnabledFeature: true,
      active: false,
    },
    {
      id: 3,
      title: 'Liste des inscrits',
      link: '/members',
      hasEnabledFeature: true,
      active: false,
    },
    {
      id: 2,
      title: 'Lien externe',
      link: '/www.google.fr',
      hasEnabledFeature: true,
      active: false,
    },
  ],
  active: false,
}
