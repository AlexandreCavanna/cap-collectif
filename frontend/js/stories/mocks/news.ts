import { author } from './users'

export const post = {
  title: 'Je suis le titre',
  url: '#',
  image: {
    src: 'https://source.unsplash.com/random/150x100',
    alt: 'news picture',
  },
  body: ' Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
  date: {
    released: '2019-11-09',
  },
  theme: 'Test & Democratie',
  user: author,
}
export const withoutParaph = {
  title: 'un titre pour le deuxi√®me cas',
  url: '#',
  image: {
    src: 'https://source.unsplash.com/random/150x100',
    alt: 'visual for other case',
  },
  date: {
    released: '19/12/2019',
  },
  theme: 'Test et Jest',
  user: author,
  body: '',
}
export default post
