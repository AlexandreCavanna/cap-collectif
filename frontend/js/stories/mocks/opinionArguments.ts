import { author } from './users'

export const opinionArgument = {
  id: 1,
  user: author,
  trashedStatus: '',
  createdAt: '1 mars 2016 à 00:00',
  publishedAt: '1 mars 2016 à 00:00',
  published: true,
  related: {
    url: 'https://ui.cap-collectif.com/',
    title: 'Opinion XXX',
  },
  body: 'In placeat reiciendis ut. Officiis praesentium quia minima ut tenetur officiis. Eaque fugit voluptates temporibus suscipit provident culpa culpa. Magni recusandae dolorem aut id.',
  votes: {
    totalCount: 50,
  },
  viewerHasVote: false,
  contribuable: true,
  reported: false,
}
export const opinionArguments = [
  { ...opinionArgument, id: 1 },
  { ...opinionArgument, id: 2, user: { ...opinionArgument.user, vip: true } },
  { ...opinionArgument, id: 3, user: null },
  { ...opinionArgument, id: 4, trashedStatus: 'INVISIBLE', body: null },
  { ...opinionArgument, id: 5, published: false },
  { ...opinionArgument, id: 6, viewerHasVote: true },
  { ...opinionArgument, id: 7, contribuable: false },
  { ...opinionArgument, id: 8, viewerHasVote: true, contribuable: false },
  { ...opinionArgument, id: 9, user: { ...opinionArgument.user, isViewer: true } },
  { ...opinionArgument, id: 10, reported: true },
]
