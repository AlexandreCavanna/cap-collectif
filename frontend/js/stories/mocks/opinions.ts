import { author } from './users'

export const opinion = {
  id: 'opinion1',
  url: 'https://ui.cap-collectif.com/',
  title: 'Opinion',
  user: author,
  trashedStatus: '',
  createdAt: '1 mars 2018',
  publishedAt: '1 mars 2018',
  published: true,
  pinned: false,
  votes: {
    totalCount: 8,
  },
  versions: {
    totalCount: 3,
  },
  arguments: {
    totalCount: 4,
  },
  sources: {
    totalCount: 0,
  },
  votesMitige: {
    totalCount: 1,
  },
  votesNok: {
    totalCount: 3,
  },
  votesOk: {
    totalCount: 4,
  },
  previewVotes: new Array(8).fill(author),
  body: 'Lorem ipsum dolor amet direct trade cornhole exercitation hashtag mixtape, master cleanse vexillologist. Irure kickstarter single-origin coffee fashion axe vexillologist viral. Selfies etsy cloud bread heirloom, sunt craft beer fixie palo santo asymmetrical.',
  project: {
    opinionCanBeFollowed: true,
  },
  section: {
    voteWidgetType: 2,
  },
  appendices: [],
  answer: null,
}
export const opinions = [
  { ...opinion },
  { ...opinion, user: { ...opinion.user, vip: true } },
  { ...opinion, user: null },
  {
    ...opinion,
    votes: {
      totalCount: 0,
    },
  },
  { ...opinion, updatedAt: '15 mars 2018' },
  { ...opinion, updatedAt: '15 mars 2018', pinned: true },
  { ...opinion, updatedAt: '15 mars 2018', ranking: 'Label' },
  { ...opinion, updatedAt: '15 mars 2018', pinned: true, ranking: 'Label' },
  { ...opinion, ranking: 'Label' },
  { ...opinion, trashedStatus: 'INVISIBLE' },
  { ...opinion, pinned: true, ranking: 'Label' },
  { ...opinion, pinned: true },
  { ...opinion, published: false },
]
