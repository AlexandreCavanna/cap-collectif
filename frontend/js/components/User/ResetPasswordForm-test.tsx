/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { ResetPasswordForm } from './ResetPasswordForm'
import { formMock } from '~/mocks'

describe('<ResetPasswordForm />', () => {
  const props = { ...formMock, token: 'ken-le-token', dispatch: jest.fn() }
  it('should render correctly', () => {
    const wrapper = shallow(<ResetPasswordForm {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
