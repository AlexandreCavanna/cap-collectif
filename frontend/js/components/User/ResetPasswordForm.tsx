import React from 'react'
import { Field, reduxForm, SubmissionError, submit } from 'redux-form'
import { FormattedMessage } from 'react-intl'
import type { StyledComponent } from 'styled-components'
import styled from 'styled-components'
import { Panel } from 'react-bootstrap'
import UserPasswordField from '~/components/User/UserPasswordField'
import component from '~/components/Form/Field'
import type { Dispatch } from '~/types'
import { asyncPasswordValidate } from '~/components/User/UserPasswordComplexityUtils'
import SubmitButton from '~/components/Form/SubmitButton'
import ResetPasswordMutation from '~/mutations/ResetPasswordMutation'

export type Props = ReduxFormFormProps & {
  token: string
}
type FormValues = {
  token: string
  new_password: string
  new_password_confirmation: string
}
export const validate = (values: FormValues) => {
  const errors: any = {}

  if (!values.new_password && !values.new_password_confirmation) {
    return {}
  }

  if (!values.new_password || values.new_password.length < 1) {
    errors.new_password = 'at-least-8-characters-one-digit-one-uppercase-one-lowercase'
  }

  if (!values.new_password_confirmation || values.new_password_confirmation.length < 1) {
    errors.new_password_confirmation = 'fos_user.password.mismatch'
  }

  if (
    values.new_password &&
    values.new_password_confirmation &&
    values.new_password_confirmation !== values.new_password
  ) {
    errors.new_password_confirmation = 'fos_user.password.mismatch'
  }

  return errors
}

const onSubmit = (values: FormValues, dispatch: Dispatch, { intl, token }) => {
  const input = {
    token,
    password: values.new_password,
  }
  return ResetPasswordMutation.commit({
    input,
  }).then(response => {
    if (!response.resetPassword || !response.resetPassword.user || response.resetPassword.error) {
      if (response.resetPassword && response.resetPassword.error) {
        throw new SubmissionError({
          _error: response.resetPassword.error,
        })
      } else {
        throw new SubmissionError({
          _error: intl.formatMessage({
            id: 'global.error.server.form',
          }),
        })
      }
    }

    if (response.resetPassword.user) {
      window.location.href = '/'
    }
  })
}

export const formName = 'recreate_password_form'
const Container: StyledComponent<any, {}, HTMLDivElement> = styled.div`
  width: 50%;
  margin: 20px auto;

  .reset-password-form {
    margin: auto;
  }

  .flex-column {
    display: flex;
    flex-direction: column;
  }
`
export const ResetPasswordForm = ({ submitting, pristine, dispatch, invalid, handleSubmit }: Props) => {
  return (
    <Container>
      <form className="reset-password-form" onSubmit={handleSubmit}>
        <Panel>
          <Panel.Body>
            <div className="flex-column ml-10 mr-10">
              <label htmlFor="password-form-new">
                <FormattedMessage id="change_password.form.password" />
              </label>
              <UserPasswordField formName={formName} id="password-form-new" name="new_password" />
              <label htmlFor="password-form-confirmation">
                <FormattedMessage id="change_password.form.password_confirmation" />
              </label>
              <Field
                type="password"
                component={component}
                name="new_password_confirmation"
                id="password-form-confirmation"
              />
              <SubmitButton
                label="global.confirm"
                bsStyle="success"
                id="reset-content-confirm"
                disabled={pristine || submitting || invalid}
                isSubmitting={submitting}
                onSubmit={() => dispatch(submit(formName))}
              />
            </div>
          </Panel.Body>
        </Panel>
      </form>
    </Container>
  )
}

const asyncValidate = (values: FormValues, dispatch: Dispatch) => {
  return asyncPasswordValidate(formName, 'new_password', values, dispatch)
}

export default reduxForm({
  onSubmit,
  validate,
  asyncValidate,
  form: formName,
})(ResetPasswordForm)
