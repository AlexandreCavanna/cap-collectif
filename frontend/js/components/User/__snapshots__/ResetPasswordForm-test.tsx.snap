// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResetPasswordForm /> should render correctly 1`] = `
<ResetPasswordForm__Container>
  <form
    className="reset-password-form"
    onSubmit={[MockFunction]}
  >
    <Uncontrolled(Panel)>
      <PanelBody
        bsClass="panel"
        collapsible={false}
      >
        <div
          className="flex-column ml-10 mr-10"
        >
          <label
            htmlFor="password-form-new"
          >
            <FormattedMessage
              id="change_password.form.password"
              values={Object {}}
            />
          </label>
          <Connect(UserPasswordField)
            formName="recreate_password_form"
            id="password-form-new"
            name="new_password"
          />
          <label
            htmlFor="password-form-confirmation"
          >
            <FormattedMessage
              id="change_password.form.password_confirmation"
              values={Object {}}
            />
          </label>
          <Field
            component={[Function]}
            id="password-form-confirmation"
            name="new_password_confirmation"
            type="password"
          />
          <Connect(SubmitButton)
            bsStyle="success"
            disabled={true}
            id="reset-content-confirm"
            isSubmitting={false}
            label="global.confirm"
            onSubmit={[Function]}
          />
        </div>
      </PanelBody>
    </Uncontrolled(Panel)>
  </form>
</ResetPasswordForm__Container>
`;
